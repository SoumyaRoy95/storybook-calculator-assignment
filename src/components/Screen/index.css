.input {
  height: 50px;
  width: 139px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  font-weight: bold;
  font-size: 16px;
  background-color: black;
  color: white;
  padding: 10px;
  outline: 1px solid brown;
}
