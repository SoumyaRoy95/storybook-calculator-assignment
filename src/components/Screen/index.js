import React from "react";
import "./index.css";

const Screen = (props) => {
  const { input, ...rest } = props;
  return (
    <div className="input" {...rest}>
      {input}
    </div>
  );
};

export default Screen;
