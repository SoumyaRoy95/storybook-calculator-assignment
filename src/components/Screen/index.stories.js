import React from "react";
import Screen from "./index";

export default {
  title: "Screen",
  component: Screen,
};

export const Result = () => <Screen input="1"></Screen>;
