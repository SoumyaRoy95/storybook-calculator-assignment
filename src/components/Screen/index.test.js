import Screen from "./index";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  render(<Screen input="0" />);
});

it("renders screen correctly", () => {
  render(<Screen input="1+1" />);
  expect(screen.getByText("1+1")).toBeInTheDocument();
});

it("matches screenshot", () => {
  const file = renderer.create(<Screen input="1" />).toJSON();
  expect(file).toMatchSnapshot();
});
