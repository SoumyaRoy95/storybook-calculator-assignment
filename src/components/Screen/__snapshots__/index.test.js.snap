// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches screenshot 1`] = `
<div
  className="input"
>
  1
</div>
`;
